### DISCUSSION TITLE
Better Text-to-speech

### DISCUSSION THREAD

**Discussion by delightedbydhamma** [delightedbydhamma/impossible-pali-project#2] [Ideas]
At the moment all the Text-to-Speech examples available in Pali are quite stilted and sound a lot like a list of words being read out. I'm creating this thread so we can document some ideas about how to create better emphasis and linking(?) between words.

**Comment by delightedbydhamma** [comment]
What is X-SAMPA? It stands for Extended Speech Assessment Methods Phonetic Alphabet.
[Check out details and the chart here](https://chromium.googlesource.com/chromiumos/third_party/espeak-ng/+/HEAD/docs/phonemes/xsampa.md#vowels)

Here are some initial thoughts which I need to test out in Polly or some other TTS engine

Split to syllables (Using Legacy sc  k-a)
Add stress marks as per underlining (using X-SAMPA)
Convert syllables to X-SAMPA
Find syllables which end in long vowels o & e and lengthen them (add ::)
Shorten long vowels before double consonants 
Nasalise vowels followed by ṃ and lengthen
Clip short vowels at end of words and tighten space after the word
At the end of words, add a bit more space after long vowels or ṃ which occur at the end of words

**Reply by GillianPerrett** [reply]
Re Split to syllables (Using Legacy sc k-a)
Why syllables rather than phonemes?

**Reply by delightedbydhamma** [reply]
because the syllables have the little underline marks for stress. 
We need to capture the stress and the phonemes

**Reply by GillianPerrett** [reply]
In which versions of the text doe we see "the little underline marks for stress'?

**Reply by delightedbydhamma** [reply]
go to legacy.suttacentral.net, choose a sutta in pali, click the three line (hamburger) menu. Click Controls k-a and you get it broken up.
![Screen Shot 2023-03-18 at 3 11 49 pm](https://user-images.githubusercontent.com/105905254/226084271-d09a48f0-4c50-4813-b602-a19a2557a63d.png)

**Reply by GillianPerrett** [reply]
Gotcha now.  This is fantastic - why didn't I know about it before?  and why was it dropped from SC?
It seems to assume that syllable stress within words is invariable -- fair enough since we have no recordings from the era, and **very** convenient for our purposes.  :)

**Reply by delightedbydhamma** [reply]
I showed it to at Dhammasara one day :)
I don't know if it's 100% correct though because sometimes the only 3 letter syllable is -taṃ etc and really surely the emphasis should be on the stem at that point. The -aṃ would be the secondary stress. 

"bud.%dhaṃ
pa.ri."sud.%dhaṃ
"pac.ca."vek.khi.%tab.baṃ <-- a pet peeve of certain people at DS is the emphasis on the 'bum'

Symbol | Name
-- | --
"◌ | primary stress
%◌ | secondary stress

**Reply by GillianPerrett** [reply]
Showing a geriatric memory things doesn't stick.:(. Unfortunately words have to be sent!!!
But actually I meant **why didn't my Pali teachers 2 years ago show me this**
Where the emphasis "should" be is a mute point; we have no way of knowing how the ancient Indians sounded.
**However I agree, that to make sense to our ears that would be a great help.**

**Reply by GillianPerrett** [reply]
I am thinking as deeply as I can about this issue of a balance between what we actually know about Pali and what sounds acceptable to western ears vs S Asian ears vs E Asian ears.

**Reply by delightedbydhamma** [reply]
This, to me is a beautiful balance https://forvo.com/user/sameeruddowlakhan/pronounced-words/pi/ 
I wonder of Michael can make an AI based on it?

**Comment by delightedbydhamma** [comment]
I think our efforts with Amazon have paid off.
Using one of their 'neural' voices and lots of lexicon modifications I have produced some pretty coherent Pali.
Check the [audio folder/phrases](https://github.com/delightedbydhamma/impossible-pali-project/tree/audio/audio_samples/phrases) in the audio branch (I'm still learning github)

taṃ kiṃ maññasI?

**Comment by delightedbydhamma** [comment]
Wasn't sure where to put this... but this is some pretty nice, clean pali pronunciation!
https://www.youtube.com/watch?v=gJC6iiQCDkY (the script is Brahmi- whatever).

**Comment by delightedbydhamma** [comment]
As per our email conversation I fixed pha in pa_pha_ba_bha_ma https://github.com/delightedbydhamma/impossible-pali-project/blob/audio/audio_samples/syllables/
